### ISSUE TITLE
Twidere only showing Twitter mentions in Notifications tab, not likes, follows or retweets

### ISSUE THREAD

**Issue by fschaap** [TwidereProject/Twidere-Android#941]
Am I going crazy or did the Notifications tab actually show likes, follows and retweets until recently? Right now the Notification tab is only showing mentions for Twitter.

## Expected behavior
Notification tab shows all the notifications that the Twitter web client also shows when the "All" option is selected.

## Actual behavior
Notification tab only shows mentions.

## Steps to reproduce
Yes, I have been through all the settings.No, I have not checked 'Only mentions' in the tab settings. I have enabled, turned off and enable streaming again. I have tried the default Twidere consumer key/secret and the official Twitter key/secret. I tried everything I could think of and I have even removed the accounts, cleared storage, uninstalled the app, rebooted phone, reinstalled app, added accounts. But no dice.

Reproduce: install Twidere, add account, go to Notifications tab and see if it shows likes and retweets of your tweets.

## Extra info

Android version: 7.1.1

Network type:

- [x] Cellular
- [x] Wi-Fi

App version: (version code here)

Build variant: 3.6.24

- [x] Google
- [ ] F-Droid

Micro-blogging service:

- [x] Twitter
- [ ] Fanfou
- [ ] GNU Social / StatusNet
- [ ] Mastodon

App settings:

- [ ] Custom API
- [x] Custom consumer key/secrets
- [x] Official keys
- [ ] Using [Thumbor](https://github.com/thumbor/thumbor)
- [ ] Connected through proxy
- [x] Media preview enabled

**Comment by fschaap** [comment]
I forgot to mention: likes, follows and re-toots are shown in the Notifications tab for my Mastodon account. Just not for Twitter any more.

**Comment by Yowlen** [comment]
I've personally had no problems using the consumer/secret keys for Twitter's official Android app, but this isn't the first time someone's complained about such an issue. I've seen some past reports here saying the same thing.

A few things to make sure of: Twidere has both a "Twidere" and "User settings" mode when adding Twitter accounts. You'll wanna select "User settings" & then use the button in the top-left to change the keys before hitting the sign-in button - unless you've already changed the defaults used for the "Twidere" one in the advanced settings area. I can't personally vouch for changing the defaults, since I don't use that method, but I've never had any issues using "User settings" & manually changing the keys each time.

**Comment by yiskah** [comment]
I've been getting this same issue for a long while now.

**Comment by kuba-orlik** [comment]
Me too! It's been like that since I use it, I though it's just not implemented or smth

**Comment by novadeviator** [comment]
i had this same problem and I thought that it is fixed if you have streaming enabled but i get so little retweets and follows and that i'm not sure if it works.

**Comment by fschaap** [comment]
We've had a couple of updates since I filed this ticket. I have completely wiped and reinstalled the app and I have streaming enabled. Now I see all types of notifications.

However, I used to get all types of notifications without enabling streaming. I will disable streaming and see what happens the next couple of days.

**Comment by yiskah** [comment]
Even with streaming enabled I only get my mastodon notifications + twitter mentions. No twitter RTs, follows etc.

**Comment by Yowlen** [comment]
I'd like to try to narrow down the methods by which you're adding the accounts. As I've said above, I've never had any problems with my method, so if we can see where we do things differently, maybe we can figure out where things are going wrong.

When I add my Twitter account, here's the exact steps I follow:
1. If I'm re-adding an account, I delete the current one from Twidere's Account page, just in case. [Mariotaku's own instructions](https://gist.github.com/mariotaku/5465786) say it shouldn't be necessary, but better safe than sorry, imo.
2. When I add a new account, I select `User Settings` from the Twitter dropdown box. 
![screenshot_2017-10-26-10-11-56](https://user-images.githubusercontent.com/17078692/32058228-7824c07c-ba37-11e7-8fc1-6b8ccff54000.png)
3. Then I go into the API settings using the icon in the top-right, leaving everything at its defaults except for the consumer/secret keys. (You might wanna use the `Load Defaults` option at the top just to be sure, or you can line them up with my screenshot below.)
![screenshot_2017-10-26-10-12-23](https://user-images.githubusercontent.com/17078692/32058916-924bcb70-ba38-11e7-9fe0-f45768d794aa.png)
4. As for the consumer/secret keys themselves, I use the Twitter for Android ones from [this pastebin](https://pastebin.com/Fgqed59f). I remember having issues with the ones on Mariotaku's instruction page in the past even though they look identical. I don't know if it was a Unicode issue, or formatting, or if I just imagined it, but I know the pastebin keys work, so I haven't gone back to verify.
**Important note:** Make sure there's no spaces before or after the keys! Certain keyboards/Android versions will put spaces before or after the pasted text.
5. After saving the API settings, sign into the account. It should say you're authorizing the official Twitter client on the login screen.

If any of these steps differ from yours, explaining how may help to get to the bottom of the problem.

**Comment by yiskah** [comment]
Yeah that's not the method I use I used the simple OAuth method which booted me to twitter to log in and then came back. I think? that's what I recall.

**Comment by Yowlen** [comment]
The simple OAuth method uses the default Twidere keys. That's why it doesn't work for you. You need to either make some custom keys at dev.twitter.com or use the official client's keys as per the links above. The custom keys are Twitter's "official" way of doing things. Using the keys from Twitter's official clients in 3rd party apps technically goes against their ToS, but they don't really enforce it, I think.

Anyway, you'll wanna switch to the method I explained above & enter the keys of your choice. You should be good to go then.

**Comment by kkarhan** [comment]
same problem. will watch this.topic...

**Comment by Yowlen** [comment]
@kkarhan What's the problem? As I said above, listing the steps you took to enter the consumer/secret keys can help narrow down where things are going wrong.

**Comment by thecosmicyes** [comment]
@Yowlen thank you so much! I nearly got mad on that issue.

**Comment by Yowlen** [comment]
I get likes too, yes. I'm not sure what could be the cause if you're only
getting some of it.

Which key are you using? As I stated above, I'm using the one for Twitter's
Android app. If you're using a different one, maybe that client has a
different way of doing things, which makes it incompatible with the Android
way.

>
Beyond that, all I can figure is to make sure all the settings are correct.
That includes the settings in Twidere's Notifications section, which I know
can disable some of them. "Mentions only" should be off. "Your following
only" may be causing issues too if the likes are coming from non-followers
& the retweets were from followers.

As a last resort, you can load up Twitter's desktop page in a browser &
check the notifications settings there to make sure everything's set right
there too.

**Comment by ajoah** [comment]
@Yowlen Oops sorry, i think i was wrong tab. (My app is in french). I talk about home tab, and this topic talk about the notifications tab ("interactions" tab in french ?). On this tab, no problem i have likes.

_It is off topic, but if you know how to get following likes on home tab (like the official app), I'll take it._

**Comment by yogubert** [comment]
I recently changed my cellphone and also started having this issue (not showing likes and RTs, only mentions) as soon as I installed Twidere again. Maybe it has something to do with the android version being different?

**Comment by Yowlen** [comment]
@yogubert Please check to ensure all the API settings line up properly. You can use my tutorial above for a list of all the proper settings, as well as step-by-step instructions on how to enter them.

If you're still having issues after verifying that, then idk what to say. I've used Twidere on every Android version from Kit Kat to the latest Oreo. My Nougat one used Samsung TouchWiz while all the rest were AOSP-based. Every time, the official Twitter for Android keys worked flawlessly.

In essence, the Android version can't cause this kind of problem because it's an issue with Twitter's API authorizations, not an issue with Android's functionality.

**Comment by trwnh** [comment]
Official Twitter API now respects the notification settings of the official app. If you disabled a certain type of notification in an official Twitter app, then your account won't get those notifications via the API anymore.

I had a similar thing happen to me with Flamingo for Twitter -- I started receiving only mentions, because I had disabled everything except mentions in the Twitter for Android app before uninstalling it.

**Comment by xa2ds** [comment]
It has worked for me!!! Thank you @Yowlen !!

**Comment by evolet95** [comment]
Hello @Yowlen, it has worked for me the first time i did, but after i put iphone key and secret and the notifications does not work again. I put android key again but this time it does not work anymore. Any suggest?

**Comment by Yowlen** [comment]
Did you try removing the account in Twidere and then re-adding it with the Android keys? If it still doesn't work, you can try uninstalling and reinstalling Twidere.

If that still doesn't work, I'm afraid I can't really help you.

**Comment by evolet95** [comment]
@Yowlen Yes i tried that. when i put the android keys and select "user settings" to log-in it does not do  nothing, i mean, when i use android keys i can't login.

**Comment by evolet95** [comment]
i've tried on different android devices,  put in oficial keys but i cant login. (nothing happens after pressing user settings), it seems like twitter changed something.

**Comment by evolet95** [comment]
i've tried on different android devices,  put in oficial keys but i cant login. (nothing happens after pressing user settings), it seems like twitter changed something.

**Comment by evolet95** [comment]
@Yowlen Hey, i found a way: Set official keys again and chance the method of authentication from 0Auth to xAuth.

**Comment by Tedko** [comment]
im not sure if this is possible with the current limited twitter API -

**Comment by wheelsofsteel** [comment]
Can confirm this is still an issue. I've used the twitter android keys from the pastebin listed above and also here: https://gist.github.com/sokcuri/42fba0b455a7b7b8ab772ba344268def

Unfortunately I either get a '403' error or it says the Twitter account has been suspended.

I remember it worked fine before with the method listed above but unfortunately not now.

**Comment by Tedko** [comment]
@wheelsofsteel can you try test again? Thx

**Comment by kr428** [comment]
Same problem here, recent Twidere build from fdroid or Google Play (Beta). Notifications flawlessly work on Mastodon; on Twitter I only see notifications for comments or DMs, no stars/retweets of things I posted.

**Comment by PLAYERUNKNOWNHatsune01** [comment]
The Issue still exist tho! Any hot fix/update to purge this problem?

**Comment by strugee** [comment]
> The Issue still exist tho!

That's why this GitHub issue is still open. If anyone knew how to fix this permanently, it would be mentioned here. You could try reading through the thread and seeing if some of the things people have tried work for you.

**Comment by guttermonk** [comment]
+1
